internal/provider/resource_bytes.go

```diff
@@ -0,0 +1,187 @@
+	resp.TypeName = req.ProviderTypeName + "_bytes"
+}
+
+func (r *bytesResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = bytesSchemaV0()
+}
+
+func (r *bytesResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan bytesModelV0
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	bytes := make([]byte, plan.Length.ValueInt64())
+	_, err := rand.Read(bytes)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Create Random bytes error",
+			"There was an error during random generation.\n\n"+
+				diagnostics.RetryMsg+
+				fmt.Sprintf("Original Error: %s", err),
+		)
+		return
+	}
+
+	u := &bytesModelV0{
+		ID:           types.StringValue("none"),
+		Length:       plan.Length,
+		ResultBase64: types.StringValue(base64.StdEncoding.EncodeToString(bytes)),
+		ResultHex:    types.StringValue(hex.EncodeToString(bytes)),
+		Keepers:      plan.Keepers,
+	}
+
+	diags = resp.State.Set(ctx, u)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read does not need to perform any operations as the state in ReadResourceResponse is already populated.
+func (r *bytesResource) Read(context.Context, resource.ReadRequest, *resource.ReadResponse) {
+}
+
+// Update is intentionally left blank as all required and optional attributes force replacement of the resource
+// through the RequiresReplace AttributePlanModifier.
+func (r *bytesResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var model bytesModelV0
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &model)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, &model)...)
+}
+
+// Delete does not need to explicitly call resp.State.RemoveResource() as this is automatically handled by the
+// [framework](https://github.com/hashicorp/terraform-plugin-framework/pull/301).
+func (r *bytesResource) Delete(context.Context, resource.DeleteRequest, *resource.DeleteResponse) {
+}
+
+func (r *bytesResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	bytes, err := base64.StdEncoding.DecodeString(req.ID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Import Random bytes Error",
+			"There was an error during the parsing of the base64 string.\n\n"+
+				diagnostics.RetryMsg+
+				fmt.Sprintf("Original Error: %s", err),
+		)
+		return
+	}
+
+	var state bytesModelV0
+
+	state.ID = types.StringValue("none")
+	state.Length = types.Int64Value(int64(len(bytes)))
+	state.ResultBase64 = types.StringValue(req.ID)
+	state.ResultHex = types.StringValue(hex.EncodeToString(bytes))
+	state.Keepers = types.MapValueMust(types.StringType, nil)
```

Comment:
I think perhaps we want to use `state.Keepers = types.MapNull(types.StringType)` here otherwise there will be a forced update following import if the configuration does not contain an empty `keepers` attribute. For example,

```terraform
resource "random_bytes" "basic" {
  length = 16
  keepers = {}
}
```

Could you add a test to ensure that following import when apply is run there are no planned changes?

Reply:
I added a test, `types.MapNull(types.StringType)` does not seems to changes the outcome, but with the test I think it's validating this scenario